src/client/extension.ts

```diff
@@ -89,45 +123,64 @@ export async function activate(
         vscode.workspace.onDidChangeTextDocument(() => {
             if (vscode.window.activeTextEditor === undefined) {
                 return;
-            } else if (
-                isCurrentDocumentEdited()
-            ) {
+            } else if (isCurrentDocumentEdited()) {
                 if (PortalWebView?.currentPanel) {
-                    _telemetry.sendTelemetryEvent('PortalWebPagePreview', { page: 'ExistingPage' });
+                    _telemetry.sendTelemetryEvent("PortalWebPagePreview", {
+                        page: "ExistingPage",
+                    });
                     PortalWebView?.currentPanel?._update();
                 }
             }
         })
     );
 
     if (vscode.window.registerWebviewPanelSerializer) {
-        vscode.window.registerWebviewPanelSerializer(
-            PortalWebView.viewType,
-            {
-                async deserializeWebviewPanel(webviewPanel: vscode.WebviewPanel) {
-                    PortalWebView.revive(webviewPanel);
-                },
-            }
-        );
+        vscode.window.registerWebviewPanelSerializer(PortalWebView.viewType, {
+            async deserializeWebviewPanel(webviewPanel: vscode.WebviewPanel) {
+                PortalWebView.revive(webviewPanel);
+            },
+        });
     }
-
-    const areCRUDoperationEnabled = vscode.workspace.getConfiguration(SETTINGS_EXPERIMENTAL_STORE_NAME).get(PORTAL_CRUD_OPERATION_SETTING_NAME);
+    const areCRUDoperationEnabled = vscode.workspace
+        .getConfiguration(SETTINGS_EXPERIMENTAL_STORE_NAME)
+        .get(PORTAL_CRUD_OPERATION_SETTING_NAME);
     if (areCRUDoperationEnabled) {
         // Add CRUD related callback subscription here
         await handleFileSystemCallbacks(_context);
     }
 
-    const cliContext = new CliAcquisitionContext(_context, _telemetry)
+    const cliContext = new CliAcquisitionContext(_context, _telemetry);
     const cli = new CliAcquisition(cliContext);
     const cliPath = await cli.ensureInstalled();
     _context.subscriptions.push(cli);
     _context.subscriptions.push(new PacTerminal(_context, _telemetry, cliPath));
-
-    const workspaceFolders = vscode.workspace.workspaceFolders?.map(fl => ({ ...fl, uri: fl.uri.fsPath } as WorkspaceFolder)) || []
+    const workspaceFolders =
+        vscode.workspace.workspaceFolders?.map(
+            (fl) => ({ ...fl, uri: fl.uri.fsPath } as WorkspaceFolder)
+        ) || [];
     if (workspaceContainsPortalConfigFolder(workspaceFolders)) {
-        const generator = new GeneratorAcquisition(cliContext)
-        generator.ensureInstalled();
+        const generator = new GeneratorAcquisition(cliContext);
+        await generator.ensureInstalled();
         _context.subscriptions.push(generator);
+        const yoCommandPath = generator.yoCommandPath;
+
+        _context.subscriptions.push(
```

Comment:
move this to your own function in this file - activate function is already quite long

Reply:
in fact move this whole generator block to your own function